unit-tests/heitech.zer0mqXt.core.tests/AttributeOrConventionTests.cs

```diff
@@ -0,0 +1,139 @@
+
+            // Act
+            // does have the correct name, but not the correct attribute which has precedence 
+            var result = await client.RequestAsync<ClientServerInProcTests.Request, Response>(new ClientServerInProcTests.Request());
+
+            // Assert
+            bool wasSignaled = waitHandle.WaitOne(1000);
+            Assert.True(wasSignaled);
+            Assert.False(result.IsSuccess);
+        }
+
+        [Fact]
+        public async Task RqRep_Message_with_different_ids_Returns_Failure()
+        {
+            // Arrange
+            using var responder = _patterns.CreateResponder();
+            bool calledOnError = false;
+            var waitHandle = new ManualResetEvent(false);
+            // never be reached since parsing should fail
+            responder.Respond<Request, Response>(x => new Response(), onError: () => { calledOnError = true; waitHandle.Set(); return new Response(); });
+            using var client = _patterns.CreateClient();
+
+            // Act
+            var result = await client.RequestAsync<RequestV3, Response>(new RequestV3());
+
+            // Assert
+            bool wasSignaled = waitHandle.WaitOne(1000);
+            Assert.True(wasSignaled);
+            Assert.True(calledOnError);
+            Assert.False(result.IsSuccess);
+        }
+
+        [Fact]
+        public async Task PubSub_With_MessageId_Works()
+        {
+            // Arrange
+            using var subscriber = _patterns.CreateSubscriber();
+            using var publisher = _patterns.CreatePublisher();
+            var waitHandle = new ManualResetEvent(false);
+            subscriber.RegisterSubscriber<Request>((r) => waitHandle.Set());
+
+            // Act
+            var result = await publisher.SendAsync<RequestV2>(new RequestV2());
+
+            // Assert
+            bool wasSignaled = waitHandle.WaitOne(1000);
+            Assert.True(result.IsSuccess);
+            Assert.True(wasSignaled);
+        }
+
+        [Fact]
+        public async Task PubSub_With_Incorrect_MessageId_Fails()
+        {
+            // Arrange
+            using var subscriber = _patterns.CreateSubscriber();
+            using var publisher = _patterns.CreatePublisher();
+            var waitHandle = new ManualResetEvent(false);
+            subscriber.RegisterSubscriber<Request>((r) => { }, onError: () => waitHandle.Set());
+
+            // Act
+            var result = await publisher.SendAsync<RequestV3>(new RequestV3());
+
+            // Assert
+            bool wasSignaled = waitHandle.WaitOne(1000);
+            Assert.True(result.IsSuccess);
+            Assert.True(wasSignaled);
+        }
+
+        [Zer0mqMessage("Request-Id")]
+        private class Request
+        {
+            public int RequestNumber { get; set; }
+        }
+
+        private class Response
+        {
+            public string ResponseText { get; set; }
+        }
+
+        [Zer0mqMessage("Request-Id")]
```

Comment:
I'm not sure:
Did you inted to have the same `MessageId` as `Request` here, so that the test `RqRep_With_MessageId_Works` checks that it works with different but duck-type-compatible types when the share the same `MessageId`?

Reply:
That is exactly the intended behavior. If they do not have the same set of properties, the serialization does not work out and the XtResult.IsSuccess Property returns false.

Reply:
It does - thanks.